src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -38,10 +111,168 @@ private void SetContentDispositionHeader(ActionContext context, FileResult resul
             }
         }
 
-        private void SetContentType(ActionContext context, FileResult result)
+        private static void SetAcceptRangeHeader(ActionContext context)
         {
             var response = context.HttpContext.Response;
-            response.ContentType = result.ContentType;
+            response.Headers[HeaderNames.AcceptRanges] = AcceptRangeHeaderValue;
+        }
+
+        // Internal for testing
+        internal static PreconditionState GetPreconditionState(
+            ActionContext context,
+            RequestHeaders httpRequestHeaders,
+            DateTimeOffset? lastModified = null,
+            EntityTagHeaderValue etag = null)
+        {
+            var ifMatchState = PreconditionState.Unspecified;
+            var ifNoneMatchState = PreconditionState.Unspecified;
+            var ifModifiedSinceState = PreconditionState.Unspecified;
+            var ifUnmodifiedSinceState = PreconditionState.Unspecified;
+
+            // 14.24 If-Match
+            var ifMatch = httpRequestHeaders.IfMatch;
+            if (ifMatch != null && ifMatch.Any())
```

Comment:
For checking lastModified, the checks are trivial so I am not extracting out the method. I did for etag matching though